File: src/main/java/com/warrantybee/api/dto/request/SignUpRequest.java
package com.warrantybee.api.dto.request;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

/** DTO for user sign up request */
@Getter
@Setter
@AllArgsConstructor
public class SignUpRequest {

    /** User's first name */
    private String firstname;

    /** User's last name/surname */
    private String lastname;

    /** User's email address */
    private String email;

    /** User's password */
    private String password;

    /** user address line 1 */
    private String addressLine1;

    /** User  address line 2*/
    private String addressLine2;

    /** User state ID */
    private LocalDate dateOfBirth;

    /** User phone number */
    private String phoneNumber;

    /** User gender. The actual type is a {@code Byte} which maps to a value in the {@code Gender} enumeration. */
    private Byte gender;

    /** User region ID */
    private Long regionId;

    /** User country ID */
    private Long countryId;

    /** User city */
    private String city;

    /** User postal code */
    private String postalCode;

    /** User avatar URL */
    private String avatarUrl;

    /** User's preferred culture's identifier */
    private Long cultureId;

    /** Captcha response. */
    private String captchaResponse;
}

File: src/WarrantyBee.Application/Contracts/Identity/SignUpRequest.cs
using WarrantyBee.Application.Contracts.Common;

namespace WarrantyBee.Application.Contracts.Identity;

public class SignUpRequest : BaseRequest
{
    public string Firstname { get; set; } = string.Empty;
    public string Lastname { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
    public bool HasAcceptedTermsAndConditions { get; set; }
    public bool HasAcceptedPrivacyPolicy { get; set; }
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public string? PhoneCode { get; set; }
    public string? PhoneNumber { get; set; }
    public byte? Gender { get; set; }
    public long? RegionId { get; set; }
    public long? CountryId { get; set; }
    public string? City { get; set; }
    public string? PostalCode { get; set; }
    public string? AvatarUrl { get; set; }
    public long? CultureId { get; set; }
    public byte? AuthProvider { get; set; }
    public string? AuthProviderUserId { get; set; }
}
